components/script_layout_interface/wrapper_traits.rs

```diff
@@ -319,6 +338,10 @@ pub trait ThreadSafeLayoutNode: Clone + Copy + NodeInfo + PartialEq + Sized {
 
     #[inline]
     fn selected_style(&self, _context: &SharedStyleContext) -> Arc<ServoComputedValues> {
+        if self.is_text_node() {
```

Comment:
Hmm... Not so sure about this, but we should try to make only elements arrive here too.